### ISSUE TITLE
Regression: error messages coming from Dotty are incorrectly reported

### ISSUE THREAD

**Issue by smarter** [sbt/sbt#5058]
1. git clone and start sbt in https://github.com/lampepfl/dotty-example-project
2. Run `compile` from sbt, observe a nice message:
```scala
[warn] -- Deprecation Warning: /home/smarter/opt/dotty-example-project/src/main/scala/EnumTypes.scala:31:12 
[warn] 31 |    println(list + "\n")
[warn]    |            ^
[warn]    |method any2stringadd in object Predef is deprecated: Implicit injection of + is deprecated. Convert to String to call +
[warn] one warning found
```
3. Quit sbt, switch the version from 1.2.8 to 1.3.0 in build.properties, restart, run ;clean;compile, observe an ugly message (with off-by-ones in the line and column numbers!):
```scala
[warn] /home/smarter/opt/dotty-example-project/src/main/scala/EnumTypes.scala:30:13: method any2stringadd in object Predef is deprecated: Implicit injection of + is deprecated. Convert to String to call +
[warn]     println(list + "\n")
[warn]             ^
[warn] one warning found
```

**Comment by smarter** [comment]
Relevant code probably maybe around here: https://github.com/sbt/zinc/blob/develop/internal/zinc-compile-core/src/main/scala/sbt/internal/inc/javac/DiagnosticsReporter.scala or maybe not, who knows.

**Comment by eed3si9n** [comment]
Not sure if this is it, but for scala verify ("cutest") I ported Sourcecode to Dotty and also noticed this.

In Scala 2.x the lineImpl looks like this:

https://github.com/scala/nanotest-strawman/blob/217e388f2f5255b56005e57263e8b8f856102929/shared/src/main/scala-2/verify/sourcecode/Macros.scala#L125-L129
```scala
  def lineImpl(c: Compat.Context): c.Expr[sourcecode.Line] = {
    import c.universe._
    val line = c.enclosingPosition.line
    c.Expr[sourcecode.Line](q"""${c.prefix}($line)""")
  }
```

whereas on Dotty it looks like this:

https://github.com/scala/nanotest-strawman/blob/217e388f2f5255b56005e57263e8b8f856102929/shared/src/main/scala-3/verify/sourcecode/Macros.scala
```scala
  def lineImpl given (qctx: QuoteContext): Expr[Line] = {
    import qctx.tasty._
    val line = rootPosition.startLine + 1
    '{ Line(${Util.literal(qctx)(line)}) }
  }
```

and in the test it's called like this:

```scala
34:  val line = sourcecode.Line()
35:  test("Line") {
36:    assert(line == 34)
37:  }
```

so Scala 2.x and Sourcecode return one-based line number following human convention. On the other hand, is Scala 3.x using zero-based line number, perhaps following LSP's [Position](https://microsoft.github.io/language-server-protocol/specification#position) specification?

**Comment by smarter** [comment]
That doesn't explain why sbt is not using the correct reporter.

**Comment by eed3si9n** [comment]
> That doesn't explain why sbt is not using the correct reporter.

oh I see. I missed that the entire reporter is swapped out for some reason.

**Comment by aappddeevv** [comment]
I'm linking this to [https://github.com/lampepfl/dotty/issues/7071](https://github.com/lampepfl/dotty/issues/7071).

**Comment by smarter** [comment]
i figured it out, and the reason why makes me sad: https://github.com/sbt/zinc/pull/699

**Comment by smarter** [comment]
@eed3si9n Could you reopen this so we can make sure that the fix makes it into 1.3.3 ? (I can't reopen it myself)

**Comment by eed3si9n** [comment]
Bumped to Zinc 1.3.1 in https://github.com/sbt/sbt/pull/5183
